### PR TITLE
Implement Credential model, repository, controller

### LINKED ISSUE #1: Database setup
Implement tables from [definition scheme](https://github.com/JEToP/jetop_credentials_backend/wiki/Database-Definition) following [LB4 guidelines and tutorials](https://loopback.io/doc/en/lb4/todo-tutorial.html)

- [x] **User**
   - [x] Model
   - [x] Repository
   - [x] Controller
- [x] **Service**
   - [x] Model
   - [x] Repository
   - [x] Controller
- [x] **Credential**
   - [x] Model
   - [x] Repository
   - [x] Controller
- [x] **History**
   - [x] Model
   - [x] Repository
   - [x] Controller
-----

- [ ] Setup automigrate
- [ ] Setup autoupdate

### PR DESCRIPTION
**Pull request by Thanzex**
This PR addresses #1 by implementing the Credential model, repository and controller.
Added a fix for tsc complaining about decorators.

### CODE REVIEW

**Comment by Thanzex** [comment]
@powext Linking the issue to this PR would close it as it is accepted, i would suggest leaving it unlinked for now

**Comment by powext** [comment]
> @powext Linking the issue to this PR would close it as it is accepted, I would suggest leaving it unlinked for now

As seen in the frontend the link will not automatically close the pull-request

**Comment by marcostoduto** [comment]
Seems well done! Nice job!

File: src/models/credential.model.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  @property({
+    type: 'object',
+    postgresql:{
+      dataType: 'json'
+    }
+  })
+  backupCode?: object;
+
+  @property({
+    type: 'string',
```
Comment (andreademasi) [inline src/models/credential.model.ts:27]: Shouldn't `backupFileUrl` have a `postgresql` property specifying the data type? @marcostoduto

Comment (Thanzex) [reply src/models/credential.model.ts:27]: The postgres connector already does an implicit type conversion, so string is saved as text in postgres, you can see type mappings  [here](https://loopback.io/doc/en/lb4/PostgreSQL-connector.html#type-mapping)

Comment (marcostoduto) [reply src/models/credential.model.ts:27]: @Thanzex as far as i know text types must be defined this way using POSTGRESQL connector:
`dataType: 'text',`
[Here](https://loopback.io/doc/en/lb3/PostgreSQL-connector.html#defining-models) you can find an example.
Please check your code and previous commit!

**Pull request merged**